File: LeetCode/TopKFrqElement.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number[]}
 */
 var topKFrequent = function(nums, k) {
    
  
    let freqpair=[]
    for(let i=0;i<nums.length;i++)
        {
              let sum=0
            let obj={index:i,value:nums[i]}
            for(let j=i;j<nums.length;j++)
                {
                  
                    if(nums[i]==nums[j])
                        {
                            sum+=1
                        }

                }
                    if(findifExist(nums[i],freqpair)==false)
                       {
                             obj.freq=sum
                           freqpair.push(obj)
                       }
                      
        }
    
    freqpair.sort((a,b)=>{
        if(a.freq<b.freq)
            {
                return 1
            }
        else if(a.freq>b.freq)
            {
                return -1
            }
        else return 0
    })
    
    let output=[]

    for(let i=0;i<k;i++)
        {
            output.push(freqpair[i].value)
        }
    
    return output
   
};

function findifExist(num,arr)
{
    let found=false
    arr.forEach((pair)=>{
        if(pair.value==num)
            {
                found=true
            }
    })
    
return found
}

File: LeetCode/TopKFrqElement.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        
        arr=[[] for i in range(len(nums)+1)]
        
        nums.sort()
        count=0
        val=nums[0]
        
        for i in nums:
            if val==i:
                count+=1
            else:
                arr[count].append(val)
                val=i
                count=1
        arr[count].append(val)
        
        res=[]
        
        for j in range(len(nums),0,-1):
            for n in arr[j]:
                res.append(n)
                if len(res)==k:
                    return res
                    
        